design/BufferOperations.md

```diff
@@ -98,14 +97,22 @@ What happens with the content of mappings depends of which function was used to
 A buffer can be created already mapped:
 
 ```webidl
+interface GPUPendingBuffer {
+    readonly attribute ArrayBuffer buffer;
+    GPUBuffer finish();
+};
+
 partial interface GPUDevice {
-    (GPUBuffer, ArrayBuffer) createBufferMapped(GPUBufferDescriptor descriptor);
-    Promise<(GPUBuffer, ArrayBuffer)> createBufferMappedAsync(GPUBufferDescriptor descriptor);
+    GPUPendingBuffer createBufferWithData(GPUBufferDescriptor descriptor);
+    Promise<GPUPendingBuffer> createBufferWithDataAsync(GPUBufferDescriptor descriptor);
 };
 ```
 
-`GPUDevice.createBufferMapped` returns a buffer in the mapped state along with an write mapping representing the whole range of the buffer.
-`GPUDevice.createBufferMappedAsync` returns the same values as a promise and provides more opportunities for optimization in implementations of the API.
+`GPUDevice.createBufferWithData` returns a buffer in the mapped state along with a write mapping representing the whole range of the buffer.
```

Comment:
I think we need a new state name, since this isn't truly "mapped" anymore, and calling `unmap` instead of `finish` on this buffer is either a no-op or an error.

Reply:
You're right, I neglected to update that part yet.